examples/pytorch/adaptive_sampling/adaptive_sampling.py

```diff
@@ -0,0 +1,475 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import dgl
+import dgl.function as fn
+import numpy as np
+import scipy.sparse as ssp
+from dgl.data import citation_graph as citegrh
+import networkx as nx
+
+##load data
+data = citegrh.load_cora()
+adj = nx.adjacency_matrix(data.graph)
+# reorder
+ids_shuffle = np.arange(2708)
+np.random.shuffle(ids_shuffle)
+adj = adj[ids_shuffle, :][:, ids_shuffle]
+data.features = data.features[ids_shuffle]
+data.labels = data.labels[ids_shuffle]
+##train-test split
+train_nodes = np.arange(1208)
+test_nodes = np.arange(1708, 2708)
+train_adj = adj[train_nodes, :][:, train_nodes]
+test_adj = adj[test_nodes, :][:, test_nodes]
+trainG = dgl.DGLGraph(train_adj)
+allG = dgl.DGLGraph(adj)
+h = torch.tensor(data.features[train_nodes], dtype=torch.float64)
+test_h = torch.tensor(data.features[test_nodes], dtype=torch.float64)
+all_h = torch.tensor(data.features, dtype=torch.float64)
+train_nodes = torch.tensor(train_nodes)
+test_nodes = torch.tensor(test_nodes)
+y_train = torch.tensor(data.labels[train_nodes])
+y_test = torch.tensor(data.labels[test_nodes])
+##configuration
+n_epoch=300
+lamb = 0.5
+lr = 1e-3
+weight_decay = 5e-4
+input_size = h.shape[1]
+hidden_size = 16
+output_size = data.num_labels
+##sample size for each layer during training
+layer_sizes = [256, 256]
+batch_size = 256
+##sample size for each layer during test
+test_batch_size = 64
+avg_degree = 8
+test_layer_sizes = [test_batch_size * avg_degree for _ in range(2)]
+
+
+class Sampler(object):
+    def __init__(self, graph):
+        self.graph = graph
+
+    def __len__(self):
+        raise NotImplementedError
+
+    def __iter__(self):
+        """Iterator
+
+        The iterator must return a tuple at a time.  The first element must be an array of node IDs from which
+        NodeFlow is grown.  Other elements can be arbitrary auxiliary data correspond to the current minibatch.
+        """
+        raise NotImplementedError
```

Comment:
Although the `Sampler` class is from the proposal of the new sampler refactoring, in model examples (especially when sampler refactor is not implemented) you could keep the code simple - for instance, remove this class and set the `graph` attribute in `NodeSampler` directly.